src/vs/editor/common/languages.ts

```diff
@@ -69,21 +65,30 @@ export class TokenizationResult {
 export class EncodedTokenizationResult {
 	_encodedTokenizationResultBrand: void = undefined;
 
+	constructor(
+		/**
+		 * The tokens in binary format. Each token occupies two array indices. For token i:
+		 *  - at offset 2*i => startIndex
+		 *  - at offset 2*i + 1 => metadata
+		 *
+		 */
+		public readonly tokens: Uint32Array,
+		public readonly endState: IState,
+	) {
+	}
+}
+
+export interface IBackgroundTokenizer extends IDisposable {
```

Comment:
Use `@internal` to avoid this interface showing up in monaco.d.ts, where it isn't used.

Reply:
I think it might make sense to expose this in the monaco editor though.